grovedb/src/lib.rs

```diff
@@ -306,6 +310,87 @@ impl GroveDb {
         Ok(proof_result)
     }
 
+    pub fn verify_proof(
+        path: &[&[u8]],
+        proofs: &mut Vec<Vec<u8>>, // Generic into_iterator (trait) u8
+        expected_root_hash: [u8; 32],
```

Comment:
Instead of this function, we should instead have an execute_proof function that returns the proof root hash. Then we can also have a verify proof function that uses the execute_proof function and compares the root hashes.